9-type-aliases-redux/latex/kind-judgements.tex

```diff
@@ -0,0 +1,414 @@
+	\judgbox
+	{\tyvarValid{t}}
+	{$t$ is a valid type variable} \; \\
+	$t$ is valid if it is not a builtin-type or keyword, begins with an alpha char or underscore, and only contains alphanumeric characters, underscores, and primes.
+\end{minipage}
+\\
+\\
+
+
+\begin{minipage}{\linewidth}
+	\judgbox
+	{\kformation{\Delta;\hPhi}{\hkappa}}
+	{$\hkappa$ forms a kind} \;\\
+	\begin{mathpar}
+		\inferrule[KFSing]{
+			\kindAssign{\Delta;\hPhi}{\dtau}{\Ty}
+		} {
+			\kformation{\Delta;\hPhi}{\KSing{\dtau}}
+		}
+	\end{mathpar}
+\end{minipage}
+\\
+\\
+
+\begin{minipage}{\linewidth}
+	\judgbox
+	{\kequiv{\Delta;\hPhi}{\hkappa_1}{\hkappa_2}}
+	{$\hkappa_1$ is equivalent to $\hkappa_2$} \;\\
+	\begin{mathpar}
+		\inferrule[KERefl]{ } {
+			\kequiv{\Delta;\hPhi}{\hkappa}{\hkappa}
+		}
+		\and
+		\inferrule[KESymm]{
+			\kequiv{\Delta;\hPhi}{\hkappa_1}{\hkappa_2}
+		} {
+			\kequiv{\Delta;\hPhi}{\hkappa_2}{\hkappa_1}
+		}
+		\and
+		\inferrule[KETrans]{
+			\kequiv{\Delta;\hPhi}{\hkappa_1}{\hkappa_2} \\
+			\kequiv{\Delta;\hPhi}{\hkappa_2}{\hkappa_3}
+		} {
+			\kequiv{\Delta;\hPhi}{\hkappa_1}{\hkappa_3}
+		}
+		\and
+		\inferrule[KESingEquiv]{
+			\kcequiv{\Delta;\hPhi}{\dtau_1}{\dtau_2}{\Ty}
+		} {
+			\kequiv{\Delta;\hPhi}{\KSing{\dtau_1}}{\KSing{\dtau_2}}
+		}
+	\end{mathpar}
+\end{minipage}
+\\
+\\
+
+\begin{minipage}{\linewidth}
+	\judgbox
+	{\kindAssign{\Delta;\hPhi}{\dtau}{\hkappa}}
+	{$\dtau$ is assigned kind $\hkappa$}
+	\begin{mathpar}
+		\inferrule[KAConst]{ }{
+			\kindAssign{\Delta;\hPhi}{c}{\Ty}
+		}
+		\and
+		\inferrule[KAVar]{
+			t : \hkappa \in \hPhi
+		} {
+			\kindAssign{\Delta;\hPhi}{t}{\hkappa}
+		}
+		\and
+		\inferrule[KABinOp]{
+			\kindAssign{\Delta;\hPhi}{\dtau_1}{\Ty}
+			\kindAssign{\Delta;\hPhi}{\dtau_2}{\Ty}
+		} {
+			\kindAssign{\Delta;\hPhi}{\dtau_1 \oplus \dtau_2}{\Ty}
+		}
+		\and
+		\inferrule[KAList]{
+			\kindAssign{\Delta;\hPhi}{\dtau}{\Ty}
+		} {
+			\kindAssign{\Delta;\hPhi}{\hlist{\dtau}}{\Ty}
+		}
+		\and
+		\inferrule[KAEHole]{
+			\Dbinding{u}{\hPhi'}{\hkappa} \in \Delta \\
+			\kindAssign{\Delta;\hPhi}{\sigma}{\hPhi'}
+		} {
+			\kindAssign{\Delta;\hPhi}{\hthole{u}_{\sigma}}{\hkappa}
+		}
+		\and
+		\inferrule[KANEHole]{
+			\kindAssign{\Delta;\hPhi}{\dtau}{\hkappa'} \\
+			\Dbinding{u}{\hPhi'}{\hkappa} \in \Delta \\
+			\kindAssign{\Delta;\hPhi}{\sigma}{\hPhi'}
+		} {
+			\kindAssign{\Delta;\hPhi}{\hhole{\dtau}{u}_{\sigma}}{\hkappa}
+		}
+		\and
+		\inferrule[KASelfRecognition]{
+			\kindAssign{\Delta;\hPhi}{\dtau}{\Ty}
+		}{
+			\kindAssign{\Delta;\hPhi}{\dtau}{\KSing{\dtau}}
+		}
+		\and
+		\inferrule[KASubkind]{
+			\kindAssign{\Delta;\hPhi}{\dtau}{\hkappa_1} \\
+			\kconsubkind{\Delta;\hPhi}{\hkappa_1}{\hkappa_2}
+		} {
+			\kindAssign{\Delta;\hPhi}{\dtau}{\hkappa_2}
+		}
+
+	\end{mathpar}
+\end{minipage}
+\\
+\\
+
+
+
+
+\begin{minipage}{\linewidth}
+	\judgbox
+	{\kcequiv{\Delta;\hPhi}{\dtau_1}{\dtau_2}{\hkappa}}
+	{$\dtau_1$ is equivalent to $\dtau_2$ and has kind $\hkappa_2$} \;\\
```

Comment:
same

Reply:
^^ same as my reply, I'm not sure what's missing